Sources/OpenFeature/EventHandler.swift

```diff
@@ -0,0 +1,28 @@
+import Foundation
+import Combine
+
+public class EventHandler: EventSender, EventPublisher {
+    private let eventState: CurrentValueSubject<ProviderEvent, Never>
+
+    public init(_ state: ProviderEvent) {
+        eventState = CurrentValueSubject<ProviderEvent, Never>(ProviderEvent.stale)
+    }
+
+    public func observe() -> AnyPublisher<ProviderEvent, Never> {
+        return eventState.eraseToAnyPublisher()
+    }
+
+    public func send(
+        _ event: ProviderEvent
+    ) {
+        eventState.send(event)
+    }
+}
+
+public protocol EventPublisher {
+    func observe() -> AnyPublisher<ProviderEvent, Never>
+}
+
+public protocol EventSender {
```

Comment:
Why do we need this protocol? Can this not be just a function in `EventHandler`?

Reply:
Since `EventHandler` is meant to be used by the Provider to send events, I wonder if this protocol can help testing on that side?

Reply:
I don't know... it would only help if someone is injecting a class that implements `EventSender` which I haven't seen.